transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -872,12 +892,90 @@ private AbstractWriteTask(Recycler.Handle handle) {
             super(handle);
         }
 
+        private static int estimateSize(Object o, Map<Class<?>, Integer> classSizes) {
+            int answer = 8 + estimateSize(o.getClass(), classSizes, null);
+
+            if (o instanceof ByteBuf) {
```

Comment:
@normanmaurer - We discussed supporting a potentially more flexible interface to estimate the size of an object (instead of this fixed set of conditional statements). The thought was to make the estimation more flexible in general and potentially expose the estimation to the user who may have a more accurate or faster way to estimate. WDYT?